grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx

```diff
@@ -1,342 +1,178 @@
-import React, { useCallback, useEffect, useState } from 'react';
```

Comment:
the main changes in this file were:
- made error messaging more clear/consistent across `PluginConfigPage` and `GrafanaPluginRootPage` by:
  - backend now will distinctly return 400, 401, or 403, with a specific error message
  - there was a lot of duplication of sync related functionality between these two components, this has been moved to `state/plugin`
- extracted a few portions out into separate components (ex. `ConfigurationForm`) as it was easier to test them this way